core/src/main/scala/org/apache/spark/scheduler/ReplayListenerBus.scala

```diff
@@ -40,15 +41,31 @@ private[spark] class ReplayListenerBus extends SparkListenerBus with Logging {
    *
    * @param logData Stream containing event log data.
    * @param sourceName Filename (or other source identifier) from whence @logData is being read
+   * @param sourceTruncated Indicate whether log file might be truncated (some abnormal situation 
+   *        encountered, log file not finish writing) or not
    */
-  def replay(logData: InputStream, sourceName: String): Unit = {
+  def replay(
+      logData: InputStream,
+      sourceName: String,
+      sourceTruncated: Boolean = false): Unit = {
```

Comment:
Nit: let's call this `maybeTruncated`? it's not certain that the file is truncated.